src/_posts/2017/2017-04-01-statie-generate-and-refresh.md

```diff
@@ -0,0 +1,154 @@
+
+In short - previously I was able to make [desired output](https://github.com/crazko/romanvesely.com/blob/1b9a194de34461da23545ce966c3242477466976/gulpfile.js) with [gulp-sequence](https://www.npmjs.com/package/gulp-sequence) plugin. Fortunately, I recently found there is a new version of gulp coming with some handy features, namely: **gulp.series** and **gulp.parallel** which combine multiple functions together so tasks don't have to have dependencies anymore. And we can use those features now! Please, read a [guide how to set up gulp 4][4] before you proceed.
+
+## Watching over changes
+
+Get everything needed to start:
+
+``` bash
+$ npm install --save-dev gulpjs/gulp.git#4.0 gulp-watch gulp-run browser-sync
+```
+
+Add some foundation to the `gulpfile.js`:
+
+``` javascript
+var gulp = require('gulp');
+var run = require('gulp-run');
+var browserSync = require('browser-sync').create();
+
+function generate() {
+  return run('vendor/bin/statie generate').exec();
+};
+
+function reload(done) {
+  browserSync.reload();
+  done();
+}
+```
+
+Such simple functions, aren't they? Simple, yet they do everything we need to have to set up a watcher function. Short notes on above code:
+
+- in `generate()` function `return` has to be used in order to use it in a `gulp.series()` properly
+- same applies for `done()` function used in `reload()` - repeated reloading would be not possible without it
+
+Here comes the watcher function:
+
+``` javascript
+function watch() {
+  browserSync.init({
+    server: 'output'
+  });
+
+  gulp.watch('source/**/*', gulp.series(generate, reload));
+}
+```
+
+With a use of `gulp.series(generate, reload)` we define that we want to reload the browser when new content generation finished, not before. Last step is to create a gulp task that will use previously defined functions:
+
+``` javascript
+gulp.task('default', gulp.parallel(generate, watch));
+```
+
+Now, when you run `gulp`, new content is generated by Statie along with local server running on [localhost:3000](http://localhost:3000) (default settings). Try to edit source files and save them. Browsersync should spot the changes that were generated into `output` folder and instantly refresh the browser.
+
+So we removed one step in our development process. **Content edition is no pain anymore!**
+
+## Integrate a preprocessor
+I often use Less to not get a headache with working with CSS. Let's also add it to our Statie-based project and integrate it into the generating workflow.
+
+``` bash
+$ npm install --save-dev gulp-less
+```
+
+``` javascript
+var less = require('gulp-less');
+
+function styles() {
+  return gulp.src('source/less/styles.less')
+    .pipe(less())
+    .pipe(gulp.dest('output/css/'))
+    .pipe(browserSync.stream());
+};
+```
```

Comment:
As for less, maybe it would make sense to split this post into 2.
It's quite long and I'm loosing my attention. Having 2 post with specific topic would get attention for both less (not me) and livereload groups (me) .